### ISSUE TITLE
Allow for get_correlation() to use group_by() metadata

### ISSUE THREAD

**Issue by rudeboybert** [moderndive/moderndive#38]
Update `get_correlation()` to allow for grouped correlations. Consider the following reprex of computing correlation between median income of census tract and number of Dunkin Donuts and Starbucks (currently in help file for `DD_vs_SB`):

``` r
library(moderndive)
library(dplyr)

# This works:
DD_vs_SB %>% 
  mutate(shops_per_1000 = 1000 * shops/population) %>% 
  filter(!is.na(median_income)) %>% 
  group_by(shop_type) %>% 
  summarize(cor = cor(median_income, shops_per_1000))
#> # A tibble: 2 x 2
#>   shop_type         cor
#>   <chr>           <dbl>
#> 1 dunkin_donuts -0.0575
#> 2 starbucks      0.0980

# This doesn't: 
DD_vs_SB %>% 
  mutate(shops_per_1000 = 1000 * shops/population) %>% 
  filter(!is.na(median_income)) %>% 
  group_by(shop_type) %>% 
  get_correlation(shops_per_1000 ~ median_income)
#> # A tibble: 1 x 1
#>   correlation
#>         <dbl>
#> 1   0.0000712
```
Created on 2018-12-29 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0.9000).

**Comment by rudeboybert** [comment]
Fixed in 2e75f888e50fa2985487fb56826ab2b273e1567e

**Comment by rudeboybert** [comment]
Fixed in #57
